haystack/summarizer/transformers.py

```diff
@@ -0,0 +1,137 @@
+            padding_for_one_summary: str = " ",
+    ):
+        """
+        Load a Summarization model from Transformers.
+        See the up-to-date list of available models on
+        `huggingface.co/models <https://huggingface.co/models?filter=summarization>`__
+
+        :param model_name_or_path: Directory of a saved model or the name of a public model e.g.
+                                   'facebook/rag-token-nq', 'facebook/rag-sequence-nq'.
+                                   See https://huggingface.co/models for full list of available models.
+        :param tokenizer: Name of the tokenizer (usually the same as model)
+        :param max_length: Maximum length of summarized text
+        :param min_length: Minimum length of summarized text
+        :param use_gpu: If < 0, then use cpu. If >= 0, this is the ordinal of the gpu to use
+        :param clean_up_tokenization_spaces: Whether or not to clean up the potential extra spaces in the text output
+        :param padding_for_one_summary: Padding to merge multiple docs in order to produce one summary
+        """
+
+        self.summarizer = pipeline("summarization", model=model_name_or_path, tokenizer=tokenizer, device=use_gpu)
+        self.max_length = max_length
+        self.min_length = min_length
+        self.clean_up_tokenization_spaces = clean_up_tokenization_spaces
+        self.padding_for_one_summary = padding_for_one_summary
+
+    def predict(self, documents: List[Document], generate_one_summary: bool = False, query: str = None) -> Dict:
+        """
+        Produce the summarization from the supplied documents.
+        These document can for example be retrieved via the Retriever.
+
+        :param query: Query
+        :param documents: Related documents (e.g. coming from a retriever) that the answer shall be conditioned on.
+        :param generate_one_summary: To generate single summary for all documents
+        :return: Generated answers plus additional infos in a dict like this:
+
+        ```python
+        |     {'query': 'Where is Eiffel Tower?',
+        |      'answers':
+        |          [{'query': 'Where is Eiffel Tower?',
+        |            'answer': 'The Eiffel Tower is a landmark in Paris, France.',
+        |            'meta': {
+        |                      'text': 'The tower is 324 metres ...'
+        |      }}]}
+        ```
+        """
+
+        if self.min_length > self.max_length:
+            raise AttributeError("min_length cannot be greater than max_length")
+
+        if len(documents) == 0:
+            raise AttributeError("summarizer need documents to produce summary")
+
+        contexts: List[str] = [doc.text for doc in documents]
+
+        if generate_one_summary:
+            # TODO: How to fix this?
+            # Documents order is very important to produce summary.
+            # Different order of same documents produce different summary.
+            contexts = [self.padding_for_one_summary.join(contexts)]
```

Comment:
Is this way to produce single summary is fine?

Reply:
Yes, I think it's inevitable that the generated summary depends on the order of joined docs when using these kinds of models.